pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1449,6 +1449,15 @@ private ByteBuf processMessageChunk(ByteBuf compressedPayload, MessageMetadata m
         // discard message if chunk is out-of-order
         if (chunkedMsgCtx == null || chunkedMsgCtx.chunkedMsgBuffer == null
```

Comment:
Should we handle `chunkedMsgCtx == null` in a separated block?

There are actually two cases (assuming the message is split into 2 chunks):
1. Consumer received chunk-1  before chunk-0. In this case, `chunkedMsgCtx` is null.
2. Consumer has received chunk-0, then it received chunk-0 again. In this case, `chunkedMsgCtx` is not null and `msgMetadata.getChunkId() <= chunkedMsgCtx.lastChunkedMessageId`.

The 1st case should be an unexpected case because message disordering happens. The 2nd case might be expected (I'm not sure) so that consumer should filter duplicated messages.

/cc @liangyepianzhou @poorbarcode

Reply:
There is another case:
* The consumer received chunk-0, chunk-1, and receive chunk-1 again.

Reply:
The case of `chunkedMsgCtx == null`:
**Case 1:** 
```
chunk-0 sequence ID: 1, chunk ID: 1
chunk-1 sequence ID: 1, chunk ID: 0
chunk-2 sequence ID: 1, chunk ID: 1
```
**Case 2:**
```
chunk-0 sequence ID: 1, chunk ID: 1
chunk-1 sequence ID: 1, chunk ID: 0
```
**Case 3:**
```
chunk-0 sequence ID: 1, chunk ID: 0
chunk-1 sequence ID: 1, chunk ID: 1
chunk-2 sequence ID: 1, chunk ID: 1
```
I caught `chunkedMsgCtx == null` with `msgMetadata.getChunkId() <= chunkedMsgCtx.lastChunkedMessageId` when encounter Case 3.
However, after reminders and deliberation, I think these three cases can all be processed by the original logic.

Reply:
>Consumer has received chunk-0, then it received chunk-0 again. In this case, chunkedMsgCtx is not null and msgMetadata.getChunkId() <= chunkedMsgCtx.lastChunkedMessageId.
>The 2nd case might be expected (I'm not sure) so that consumer should filter duplicated messages.

This PR does not deduplicate the chunks in a single chunk message on the broker side.
So, the 2nd case is expected. The [proposal 295](https://github.com/apache/pulsar/pull/21027) is add a patch to do deduplication of chunks on the broker side. After that the 2nd case should be an unexpected case.